pkg/commands/helpers_readwrite.go

```diff
@@ -67,6 +67,38 @@ func UpdateCluster(ctx context.Context, clientset simple.Clientset, cluster *kop
 	return nil
 }
 
+// UpdateInstanceGroup writes the updated instance group to the state store after performing validation
+func UpdateInstanceGroup(ctx context.Context, clientset simple.Clientset, cluster *kops.Cluster, allInstanceGroups []*kops.InstanceGroup, instanceGroupToUpdate *kops.InstanceGroup) error {
+	cloud, err := cloudup.BuildCloud(cluster)
+	if err != nil {
+		return err
+	}
+
+	err = cloudup.PerformAssignments(cluster, cloud)
+	if err != nil {
+		return fmt.Errorf("error populating configuration: %v", err)
+	}
+
+	assetBuilder := assets.NewAssetBuilder(cluster, "")
+	fullCluster, err := cloudup.PopulateClusterSpec(clientset, cluster, cloud, assetBuilder)
+	if err != nil {
+		return err
+	}
```

Comment:
I _think_ these can be removed because they're only operating on the Cluster object rather than the InstanceGroup we're modifying.

Reply:
I am pretty sure I need to keep this if I am going to do `DeepValidate` in the following lines. If this isn't the correct method of checking the resources though I can change it, but I didn't see another similar validation method for just instancegroups.

Reply:
Ah you're right, I missed that. thanks! I think in this case we can follow what `kops edit instancegroup` does which is largely the same:

https://github.com/kubernetes/kops/blob/97c40a3c5dd6edc952a69a307f10155391c8417d/cmd/kops/edit_instancegroup.go#L167-L183

but it only validates with `CrossValidateInstanceGroup` rather than `DeepValidate`

Reply:
Thanks for the pointer. I modified the validation call to use the same one as you referenced.